src/utilities/nep_utilities.cuh

```diff
@@ -612,11 +619,14 @@ static __device__ __forceinline__ void accumulate_f12(
   if (num_L >= L_max + 2) {
     get_f12_5body(d12, d12inv, fn, fnp, Fp[(L_max + 1) * n_max_angular_plus_1 + n], s1, r12, f12);
   }
-  s1[0] *= C3B[0];
-  s1[1] *= C3B[1];
-  s1[2] *= C3B[2];
+
   if (L_max >= 1) {
-    get_f12_1(d12inv, fn, fnp, Fp[n], s1, r12, f12);
+    float Fp_factor = 2.0f * Fp[n];
+    s1[0] *= C3B[0] * Fp_factor;
+    Fp_factor *= 2.0f;
+    s1[1] *= C3B[1] * Fp_factor;
+    s1[2] *= C3B[2] * Fp_factor;
```

Comment:
was this logic extracted out from `accumulate_f12`? Could perhaps wrap this into another function, so one doesn't have to duplicate these lines for the other `L`s?

Reply:
Yes, I have put the relevant code into a function.

Reply:
Nice!